plugins/org.jkiss.dbeaver.ui.editors.base/src/org/jkiss/dbeaver/ui/editors/DatabaseEditorInput.java

```diff
@@ -192,6 +196,17 @@ public String getDefaultFolderId()
         return defaultFolderId;
     }
 
+    @Nullable
+    @Override
+    public Color getConnectionColor() {
+        DBPDataSourceContainer container = getDataSourceContainer();
+        if (container != null) {
+            return UIUtils.getConnectionColor(container.getConnectionConfiguration());
+        } else {
+            return null;
+        }
+    }
```

Comment:
It looks exactly the same as https://github.com/dbeaver/dbeaver-ee/pull/3728/files#diff-784bb51d2aed996b869beadc48d1afdc9ecee2fcc8401f6691fb70848d66f007R105-R114
Can we avoid duplication?

Reply:
Either by providing a default implementation right in the interface, which is not ideal, as not all implementations can rely on it, or by introducing a utility function to which those two implementations will delegate. Do you think it's worth it?